### PR TITLE
Lazy payout

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

**Comment by AurevoirXavier** [comment]
@clearloop Could you fix the tests? Base on 75116bd8c45c7e46a9b8eed9cb29ee4256ed631b.

**Comment by clearloop** [comment]
@AurevoirXavier  Ok, I'll do it again, btw, you are a f**king demon.

**Comment by HackFisher** [comment]
> @clearloop Could you fix the tests? Base on 75116bd8c45c7e46a9b8eed9cb29ee4256ed631b.

@AurevoirXavier @clearloop 
Github seems to support commit to a pull request, might need some permission config from PR's owner.

https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork

BTW, the CI build failed.

**Comment by clearloop** [comment]
> Github seems to support commit to a pull request, might need some permission config from PR's owner.

Ok, I'll fork @AurevoirXavier 's branch

**Comment by AurevoirXavier** [comment]
> > @clearloop Could you fix the tests? Base on 75116bd8c45c7e46a9b8eed9cb29ee4256ed631b.
> 
> @AurevoirXavier @clearloop
> Github seems to support collaborating with others by committing to a pull request, might need some permission config from PR's owner.
> 
> [help.github.com/en/github/collaborating-with-issues-and-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork)
> 
> BTW, the CI build failed.

https://github.com/darwinia-network/darwinia/issues/347#issuecomment-599103148

**Comment by clearloop** [comment]
> Example: https://github.com/darwinia-network/darwinia/pull/170

I'm calculating `power` and `reward` now, I will end all staking tests tonight.

File: frame/staking/src/mock.rs [hunk 1]
```diff
@@ -598,3 +599,21 @@ pub fn make_all_reward_payment(era: EraIndex) {
...
+		let v: u128 = self.try_into().unwrap_or(0);
+		if v > APPROXIMATE ^ 2 {
+			v.checked_div(APPROXIMATE).unwrap_or(v)
+		} else {
+			v
+		}
+	}
+}
+
+impl Approx for u128 {}
```
Comment (clearloop) [inline frame/staking/src/mock.rs]: This `trait` terms to support fault-tolerance for two calculating way:

+ calculate with `Perbill`
+ calculate with approx values directly

**Comment by AurevoirXavier** [comment]
I'll finish the rest of the tests.

**Pull request merged**